src/idc/zephyr_idc.c

```diff
@@ -120,6 +129,18 @@ int idc_send_msg(struct idc_msg *msg, uint32_t mode)
 	int ret;
 	int idc_send_memcpy_err __unused;
 
+	if (!cpu_is_core_enabled(target_cpu)) {
+		tr_err(&zephyr_idc_tr, "Core %u is down, cannot sent IDC message", target_cpu);
+		return -EACCES;
+	}
+
+	if (zmsg->work_status == WORK_SUBMITTED) {
+		work->sync = 1;
+		ret = k_p4wq_wait(work, K_USEC(IDC_TIMEOUT));
```

Comment:
don't other `work` fields have to be set? E.g. `.handler` is called unconditionally.